src/main/java/coresearch/cvurl/io/sse/SseEventSourceBuilder.java

```diff
@@ -0,0 +1,53 @@
+package coresearch.cvurl.io.sse;
+
+import coresearch.cvurl.io.mapper.GenericMapper;
+import coresearch.cvurl.io.mapper.MapperFactory;
+import coresearch.cvurl.io.request.CVurl;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Builder class that is used to build the {@link SseEventSource} with possibility to override the default reconnection time
+ * and {@link GenericMapper} implementation that is used to parse raw data in event.
+ */
+public class SseEventSourceBuilder {
+
+    private final String url;
+    private final CVurl cVurl;
+    private int reconnectionTime = 500;
+    private GenericMapper genericMapper = MapperFactory.createDefault();
+
+    public SseEventSourceBuilder(String url, CVurl cVurl) {
+        this.url = url;
+        this.cVurl = cVurl;
+    }
+
+    /**
+     * Sets the initial reconnection time. Default is 500 milliseconds.
+     * @param reconnectionTime duration
+     * @param timeUnit time unit of the duration
+     * @return this SseEventSourceBuilder builder
+     */
+    public SseEventSourceBuilder withReconnectionTime(int reconnectionTime, TimeUnit timeUnit) {
+        this.reconnectionTime = Long.valueOf(timeUnit.toMillis(reconnectionTime)).intValue();
```

Comment:
The intValue method performs a simple casting ((int) value). You do not need to you the boxing for this. In any case, do we need to use an integer here?

Reply:
Yep, you can pass a value with the type long to the schedule method.

Reply:
Reconnection time is integer in the specification of the retry field:

> If the field value consists of only ASCII digits, then interpret the field value as an integer in base ten, and set the event stream's reconnection time to that integer.

But I guess you are right, it would look cleaner to just use long.